### PR TITLE
Add a vision inference demo

### PR DESCRIPTION
**Pull request by marbre**
This adds an example to demonstrate how to run a MNIST handwritten digit
detection vision model with IREE on bare-metal Arm.

### CODE REVIEW

**Comment by marbre** [comment]
If you want to test on a NUCLEO-F411RE, make sure to set `IREE_HAL_EXECUTABLE_LOADER_VMVX_MODULE` to `OFF`. Otherwise the executable will probably not fit into the flash.

**Comment by lucas-camp** [comment]
The python package `pillow` is a requirement for the script `build_tools/convert-image-to-c-array.py`. Should we add it to one of the requirements files?

**Comment by lucas-camp** [comment]
The output for `CMSIS` on the Corstone 300 does not seem to be correct. I got the following values:

```
Result[0] = 0.101599
Result[1] = 0.092368
Result[2] = 0.115328
Result[3] = 0.077272
Result[4] = 0.074287
Result[5] = 0.105930
Result[6] = 0.112758
Result[7] = 0.119595
Result[8] = 0.087977
Result[9] = 0.112887
Execution successful!
mnist_static_library_run_c passed
```

The correct values should be
```
Result[0] = 0.000080
Result[1] = 0.004654
Result[2] = 0.000726
Result[3] = 0.000487
Result[4] = 0.958271
Result[5] = 0.002669
Result[6] = 0.011364
Result[7] = 0.001507
Result[8] = 0.008866
Result[9] = 0.011377
Execution successful!
mnist_static_library_run_c passed
```

Setting `ARM_CPU` to `cortex-m4` in `build_tools/configure_build.sh` results in correct values, but should not be the solution for the problem.

**Comment by marbre** [comment]
> The python package `pillow` is a requirement for the script `build_tools/convert-image-to-c-array.py`. Should we add it to one of the requirements files?

I noticed this as well but was undecided on Friday. However, I agree that we should add it somewhere. I am going to add it to our main requirement file since it isn't optional when building the vision inference sample.

> The output for `CMSIS` on the Corstone 300 does not seem to be correct. I got the following values:

Thanks a lot for catching this! This is not related to the these changes but rather revealed. See #168 which fixes the issue. I will rebase after that one is merged.

**Review by lucas-camp — CHANGES_REQUESTED** [review]

File: build_tools/convert-image-to-c-array.py [hunk 1]
```diff
@@ -25,7 +26,7 @@ def serialize(input_path: str, data_format: str):
         data = np.array(image).astype(np.float32) / 255.0
         value_literal = lambda e: f"{e}f"
     cArray = f"{{{', '.join(map(value_literal, data.flat))}}}"
-    print(cArray)
+    print("const float image[] = " + cArray + ";")
```
[the remark this answers is not in the document]
Comment (marbre) [reply build_tools/convert-image-to-c-array.py]: Thanks! I took this over.

**Pull request merged**